src/main/java/javaslang/collection/HashMap.java

```diff
@@ -440,9 +440,15 @@ public int hashCode() {
     public boolean equals(Object o) {
         if (o == this) {
             return true;
-        } else if (o instanceof HashSet) {
-            final HashSet<?> that = (HashSet<?>) o;
-            return this.iterator().equals(that.iterator());
```

Comment:
@danieldietrich this calls Object.equals() but not Iterator.equals() because of well known problem with Map.Entry generics